README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# SGF

La SGF (pour *Statistique Générale de la France*) est [l’ancêtre de
l’Insee](https://www.insee.fr/fr/information/1300622), à savoir un
service ministériel chargé de collecter et analyser des statistiques
pour le gouvernement français entre 1833 et 1940.

Une partie de ces données a été saisie dans les années 1980 par une
équipe de chercheurs franco-américains à l’Inter-University Consortium
for Political and Social Research (ICPSR), situé à Ann Arbor, Michigan,
États-Unis. L’Insee a [publié
ici](https://www.insee.fr/fr/statistiques/2591397) 200 tableurs au
format dbf portant sur les recensements de la population, la démographie
et l’enseignement primaire, entre 1800 et 1925.

Le package `sgf` regroupe, normalise et documente ces indicateurs
disponibles dans la plupart des cas à l’échelon géographique du
département (des arrondissements voire des chefs-lieux
d’arrondissements dans quelques rares cas). Il permet également de
cartographier ces indicateurs au millésime géographique du département
ou de l’arrondissement correspondant au millésime de la donnée.

`sgf` est composé de :

  - 2 tables de données (format dataframe) :
      - **indicateurs\_SGF** : description des 40 000 indicateurs
        disponibles (libellé, identifiant, millésime de la donnée,
        échelles géographiques disponibles et millésime de la
        géographie)
      - **data\_SGF** : table contenant l’ensemble des données en format
        long (libellés, identifiants, millésime de la donnée et
        millésime de la géographie)
  - 2 tables de données géographiques (format sf dataframe) :
      - **geo\_DEP\_SGF\_histo** : contours géographiques des
        départements (avec leur code et libellé) selon le millésime de
        la géographie (1801/1826/1866/1876/1918, cf. "géographie
        ci-dessous)
      - **geo\_ARR\_SGF\_histo** : contours géographiques des
        arrondissements (avec leur code et libellé) selon le millésime
        de la géographie (1801/1826/1876, cf. "géographie ci-dessous)
  - 1 fonction :
      - **sgf\_sfdf** : après avoir identifié les indicateurs de son
        choix dans la table **indicateurs\_SGF**, cette fonction permet
        de créer un sf dataframe avec les indicateurs en format large

## Sources et crédits

Données statistiques : SGF, ICPSR, Insee,
[insee.fr/fr/information/1300622](https://www.insee.fr/fr/information/1300622)

Fonds cartographiques : F. Salmon, 2019,
[fondsdecarte.free.fr/](http://fondsdecarte.free.fr/)

Documentation, géoréférencement : Observatoire des Territoires.

## Installation

Le package `sgf` peut être installé depuis
[github](https://github.com/observatoire-territoires/sgf) via la
commande suivante :

``` r
remotes::install_github("observatoire-territoires/sgf")
```

Pour accéder à la documentation décrivant les tables de données, par
exemple celle listant les indicateurs :

``` r
?sgf::indicateurs_SGF
```

## Géographie

Les contours des départements de France métropolitaine ont évolué sur la
période que couvre les données de la SGF. Les 5 millésimes de cette
géographie départementale sont disponibles dans la table
`geo_DEP_SGF_histo` (idem pour les contours des arrondissements dans la
table `geo_ARR_SGF_histo`)
:

<img src="man/figures/README-plot_geographie_departement-1.png" width="100%" />

## Démo

La recherche des indicateurs se fait directement dans la table
**indicateurs\_SGF** :

``` r
library(sgf)
```

``` r
View(indicateurs_SGF)
```

Une fois le niveau géographique (variable ‘TYPE\_NIVGEO’), la source
(variable ‘TABLEAU’) et l’identifiant (variable ‘VAR\_COD’) des
indicateurs sélectionnés, la fonction **sgf\_sfdf** permet de générer un
sf dataframe des territoires avec les indicateurs correspondants, au
format large :

``` r
DEP_pop_1866 <-
  sgf_sfdf(TYPE_NIVGEO = "DEP",
           SRC ="REC_T25",
           LISTE_VAR_COD = c(215, 177, 183))

glimpse(DEP_pop_1866)
#> Observations: 89
#> Variables: 7
#> $ CODGEO                                      <chr> "01", "02", "03", ...
#> $ LIBGEO                                      <chr> "AIN", "AISNE", "A...
#> $ ANNEE_GEOGRAPHIE                            <dbl> 1866, 1866, 1866, ...
#> $ geometry                                    <GEOMETRY [m]> POLYGON (...
#> $ aveugles_total_1866                         <dbl> 318, 659, 219, 168...
#> $ personnes_sachant_lire_et_ecrire_total_1866 <dbl> 178797, 324069, 10...
#> $ total_general_de_la_population_total_1866   <dbl> 365895, 561620, 37...
```

Puis on peut cartographier le tout avec `ggplot` par exemple :

``` r
library(ggplot2)
library(scales)
library(hrbrthemes)

ggplot() +
  geom_sf(data = DEP_pop_1866 %>%
            mutate(personnes_sachant_lire_et_ecrire_total_1866_pct = 
                     personnes_sachant_lire_et_ecrire_total_1866 / total_general_de_la_population_total_1866),
          aes(fill= personnes_sachant_lire_et_ecrire_total_1866_pct),
          color = "grey20",
          lwd = 0.15) +
  scale_fill_distiller(name = "", palette = "RdBu",
                       labels = percent_format(accuracy =1),
                       direction = 1) +
  theme_ipsum() +
  theme(axis.text = element_blank(), axis.title  = element_blank(), axis.ticks  = element_blank()) +
  labs(
    title = "Part de la population sachant lire et écrire",
    subtitle = "En 1866, par département",
    caption = "Sources : Insee - SGF - ICPSR - F.Salmon - OT"
  )
```

<img src="man/figures/README-plot_exemple_carto-1.png" width="100%" />

Autre exemple avec une cartographie d’un indicateur à la maille
arrondissement en symboles proportionnels :

``` r

ARR_pop_sexe_1851 <-
  sgf_sfdf(TYPE_NIVGEO = "ARR",
           SRC ="REC_T01",
           LISTE_VAR_COD = c(11,12,13,14)) %>%
  mutate(ratio_femmes_hommes_1851 = (total_du_sexe_feminin_1851 / total_du_sexe_masculin_1851) *100  )



ggplot() +
  geom_sf(data = ARR_pop_sexe_1851, fill="grey80", color = "grey90" , lwd = 0.35) +
  geom_sf(data = geo_DEP_SGF_histo %>% filter(ANNEE_GEOGRAPHIE == 1826), 
          fill=NA, color = "grey95", lwd = 0.75 ) +
  geom_sf(data = ARR_pop_sexe_1851 %>% st_centroid(of_largest_polygon=T),
          aes(fill= ratio_femmes_hommes_1851,
              size = total_du_sexe_masculin_1851 + total_du_sexe_feminin_1851 ),
          color = "grey40", shape = 21) +
  scale_fill_gradientn(colours = c("#1b7837", "white", "#762a83"),
                       name = "Nombre de femmes\npour 100 hommes",
                       breaks = c(70,80,90,100,105,110,115),
                       values = scales::rescale(c(70, 92, 100, 104, 115))) +
  scale_size_continuous(labels=function(x) format(x, big.mark = " ", scientific = FALSE),
                        range = c(1,12),
                        name = "Population totale") +
  guides(fill = guide_legend(reverse=T)) +
  theme_ipsum() +
  theme(axis.text = element_blank(), axis.title  = element_blank(), axis.ticks  = element_blank()) +
  labs(
    title = "Sex-ratio en 1851",
    subtitle = "Par arrondissement",
    caption = "Sources : Insee - SGF - ICPSR - F.Salmon - OT"
  ) +
  theme(legend.position = c(0.95,0.5),
        axis.line=element_blank(),
        axis.title=element_blank(),
        axis.text=element_blank() )
```

<img src="man/figures/README-plot_exemple_carto_arr-1.png" width="100%" />


NEWS.md
# sgf 0.1.1

* Ajout de la maille arrondissements.

# sgf 0.1.0

* Version exploitable avec datasets et fonction.

# sgf 0.0.0.9000

* Initialisation.


R/data.R
#' Liste des indicateurs disponibles
#'
#' Description des plus de 40 000 indicateurs de la SGF publiées par l'Insee.
#'
#' @format Table (format data.frame) de 43 702 lignes et 7 variables.
#' \describe{
#'   \item{TABLEAU}{Identifiant du tableau de donnée d'origine}
#'   \item{NOM_TABLEAU}{Nom en clair du tableau de donnée d'origine}
#'   \item{VAR_COD}{Identifiant de l'indicateur (1 id unique par tableau de donnée d'origine)}
#'   \item{VAR_LIB}{Nom en clair de l'indicateur}
#'   \item{ANNEE_DONNEE}{Millésime de la donnée statistique}
#'   \item{ECHELLES_GEO}{Echelle(s) géographique(s) disponibles (DEP pour Département, 
#'   ARR pour Arrondissement, DEP_ARR pour Département+Arrondissement, CHL_VILLE pour Chef-lieu d'arrondissement)}
#'   \item{ANNEE_GEOGRAPHIE}{Millésime des contours géographiques correspondants (1801, 1826, 1866, 1876 ou 1918)}
#' }
#' @source \url{https://www.insee.fr/fr/statistiques/2591397}
"indicateurs_SGF"

#' Table des données de la SGF
#'
#' Données de la SGF (au format long) publiées par l'Insee.
#'
#' @format Table (format data.frame) de 4 627 598 lignes et 9 variables.
#' \describe{
#'   \item{NIVGEO}{Nom de l'échelle géographique (FR pour France entière, DEP pour Département, 
#'   ARR pour Arrondissement, CHL pour Chef-lieu d'arrondissement, VIL pour Ville)}
#'   \item{CODGEO}{Identifiant du territoire}
#'   \item{LIBGEO}{Libellé du territoire}
#'   \item{SRC_DATA}{Identifiant du tableau de donnée d'origine}
#'   \item{VAR_COD}{Identifiant de l'indicateur (1 id unique par tableau de donnée d'origine)}
#'   \item{VAR_LIB}{Nom en clair de l'indicateur}
#'   \item{ANNEE_DONNEE}{Millésime de la donnée}
#'   \item{ANNEE_GEOGRAPHIE}{Millésime des contours géographiques correspondants (1801, 1826, 1866, 1876 ou 1918)}
#'   \item{VAL}{Valeur de l'indicateur}
#' }
#' @source \url{https://www.insee.fr/fr/statistiques/2591397}
"data_SGF"

#' Géographies des départements  
#'
#' Contours et libellés des départements de France métropolitaine à plusieurs millésimes.
#'
#' @format Table géographique (format sf data.frame) de 437 lignes et 3 variables.
#' \describe{
#'   \item{CODGEO}{Identifiant du territoire}
#'   \item{LIBGEO}{Nom du territoire}
#'   \item{ANNEE_GEOGRAPHIE}{Millésime du contour géographique (1801, 1826, 1866, 1876 ou 1918)}
#'   \item{geometry}{Description des contours géographiques}
#' }
"geo_DEP_SGF_histo"

#' Géographies des arrondissements  
#'
#' Contours et libellés des arrondissements de France métropolitaine à plusieurs millésimes.
#'
#' @format Table géographique (format sf data.frame) de 1084 lignes et 3 variables.
#' \describe{
#'   \item{CODGEO}{Identifiant du territoire}
#'   \item{LIBGEO}{Nom du territoire}
#'   \item{ANNEE_GEOGRAPHIE}{Millésime du contour géographique (1801, 1826 ou 1876)}
#'   \item{geometry}{Description des contours géographiques}
#' }
"geo_ARR_SGF_histo"

R/sgf_sfdf.R
#' @name sgf_sfdf
#'
#' @title Créer une table de données géographiques avec des indicateurs de la SGF
#'
#' @description Générer un sf dataframe contenant les indicateurs de la SGF préalablement sélectionnés.
#'
#' @param TYPE_NIVGEO Nom court du niveau géographique ('DEP' pour département, "ARR" pour arrondissement).
#' @param SRC Nom du tableau de données dans lequel se trouvent les indicateurs à récupérer (cf. champ 'TABLEAU' dans la table 'indicateurs_SGF').
#' @param LISTE_VAR_COD Liste des codes des variables à récupérer (cf. champ 'VAR_COD' dans la table 'indicateurs_SGF').
#'
#' @return Renvoie un sf dataframe contenant code et nom du territoire ainsi que les indicateurs sélectionnés.
#'
#' @importFrom dplyr tribble filter distinct pull mutate select left_join case_when group_by summarise ungroup rename mutate_if bind_rows
#' @importFrom janitor clean_names
#' @importFrom tidyr spread gather
#' @import sf
#'
#' @examples
#' \dontrun{
#' # Creation du sf dataframe contenant les indicateurs de cheptels (bovins, ovins, etc...) 
#' issus du recensement de 1872
#' DEP_cheptels_1872 <-
#' sgf_sfdf(TYPE_NIVGEO = "DEP",
#'          SRC ="REC_T17",
#'          LISTE_VAR_COD = c(52, 59, 65, 70, 83, 87, 94))
#'
#'}
#'
#' @details
#' Deux niveaux géographiques sont disponibles : département (TYPE_NIVGEO = 'DEP') et arrondissement ("ARR"). \cr
#'
#'
#' @export

sgf_sfdf <- function(TYPE_NIVGEO, SRC, LISTE_VAR_COD) {
  
  df_data <- data_SGF %>%
    filter(NIVGEO %in% TYPE_NIVGEO) %>%
    filter(SRC_DATA %in% SRC & VAR_COD %in% LISTE_VAR_COD) %>%
    select(CODGEO,ANNEE_GEOGRAPHIE,VAR_LIB, VAL) %>%
    spread(VAR_LIB, VAL) %>%
    clean_names() %>%
    ungroup()
  
  
  annee_geo <- df_data %>% distinct(annee_geographie) %>% as.vector() %>% pull()
  
  if(TYPE_NIVGEO == "DEP") {
    data <- geo_DEP_SGF_histo %>%
      filter(ANNEE_GEOGRAPHIE %in% annee_geo) %>%
      left_join(df_data %>% select(-annee_geographie),
                by = c("CODGEO" = "codgeo")) %>%
      ungroup()
  }
  
  else {
    data <- geo_ARR_SGF_histo %>%
      filter(ANNEE_GEOGRAPHIE %in% annee_geo) %>%
      left_join(df_data %>% select(-annee_geographie),
                by = c("CODGEO" = "codgeo")) %>%
      ungroup()
  }
  
}



data-raw/clean_data.R
library(tidyverse)
library(foreign)
library(janitor)

#### fonction pour import simple d'un dataset

f_dl_import_dataset <- function(id_theme,name_theme, id_dataset){
  
  # dl
  # curl_download(paste0("https://www.insee.fr/fr/statistiques/fichier/",id_theme,"/",name_theme,"_",id_dataset,".dbf"),
  #               paste0("./data/",name_theme,"_",id_dataset,".dbf"))
  
  # import
  data = read.dbf(paste0("./data/",name_theme,"_",id_dataset,".dbf"), as.is = T)  %>%
    mutate_if(is.integer, as.numeric) 
  
}

test <- f_dl_import_dataset(id_theme <- "2653233",
                            name_theme <- "REC",
                            id_dataset <- "T02")



#### fonction pour import d'un dataset avec son année de millésime géographique correspondant

f_dl_import_md_dataset <- function(id_theme,name_theme, id_dataset, annee_recensement, annee_geographie){
  
  # dl
  # curl_download(paste0("https://www.insee.fr/fr/statistiques/fichier/",id_theme,"/",name_theme,"_",id_dataset,".dbf"),
  #               paste0("./data/",name_theme,"_",id_dataset,".dbf"))
  # 
  # import
  data = read.dbf(paste0("./data_source/",name_theme,"_",id_dataset,".dbf"), as.is = T)  %>%
    clean_names(case = "upper_camel") %>%
    mutate_if(is.integer, as.numeric) %>%
    mutate(ANNEE_GEOGRAPHIE = annee_geographie) %>%
    mutate_at(.vars = vars(-c(V1,V3,V4,V5,V6)), as.numeric)
  
  # long format
  data_lg <-
    data %>%
    gather(VAR,VAL, -c(V1,V3,V4,V5,V6,ANNEE_GEOGRAPHIE))
  
  # récup noms variables
  data_lg <-
    data_lg %>%
    left_join(liste_variables %>%
                filter(TABLEAU %in% c("Tous les tableaux",paste0(name_theme,"_", id_dataset) )) %>%
                select(VAR,VAR_LIB = LIB, ANNEE_DONNEE),
              by = "VAR") %>%
    rename(VAR_COD = VAR)
  
  # nettoyage nivgeo
  data_lg <-
    data_lg %>%
    mutate(NIVGEO = case_when(V1 %in% "0" ~ "FR",
                              V1 %in% "1" ~ "DEP",
                              V1 %in% "2" ~ "ARR",
                              V1 %in% "3" ~ "CHL",
                              V1 %in% "4" ~ "VIL",
                              V3 %in% '00' & V4 %in% '00' & V5 %in% '00' ~ "FR",
                              !V3 %in% '00' & V4 %in% '00' & V5 %in% '00' ~ "DEP",
                              !V3 %in% '00' & !V4 %in% '00' & V5 %in% '00' ~ "ARR",
                              !V3 %in% '00' & !V4 %in% '00' & !V5 %in% '00' ~ "CHL")) %>%
    mutate(CODGEO = case_when(NIVGEO %in% "FR" ~ "FR",
                              NIVGEO %in% "DEP" ~ V3,
                              NIVGEO %in% "ARR" ~ paste0(V3,V4),
                              NIVGEO %in% "CHL" ~ paste0(V3,V5),
                              NIVGEO %in% "VIL" ~ paste0(V3,V5))) %>%
    rename(LIBGEO = V6) %>%
    select(-c(V1:V5)) %>%
    mutate(SRC_DATA = paste0(name_theme,"_",id_dataset)) %>%
    select(NIVGEO, CODGEO, LIBGEO,SRC_DATA, VAR_COD, VAR_LIB, ANNEE_DONNEE,ANNEE_GEOGRAPHIE, VAL) 
  
}



#########
library(readxl)

liste_datasets <- read_xlsx("./data_source/liste_datasets.xlsx", sheet = "REC") %>%
  rbind.data.frame(read_xlsx("./data_source/liste_datasets.xlsx", sheet = "MVTPOP")) %>%
  rbind.data.frame(read_xlsx("./data_source/liste_datasets.xlsx", sheet = "TERR")) %>%
  rbind.data.frame(read_xlsx("./data_source/liste_datasets.xlsx", sheet = "ENSP"))


liste_variables <- read_xlsx("./data_source/liste_variables.xlsx", sheet = "REC") %>%
  rbind.data.frame(read_xlsx("./data_source/liste_variables.xlsx", sheet = "MVTPOP")) %>%
  rbind.data.frame(read_xlsx("./data_source/liste_variables.xlsx", sheet = "TERR")) %>%
  rbind.data.frame(read_xlsx("./data_source/liste_variables.xlsx", sheet = "ENSP"))
  

indicateurs_SGF <- liste_variables %>%
  left_join(liste_datasets %>%
              mutate(TABLEAU = paste0(name_theme,"_",id_dataset)) %>%
              select(TABLEAU,
                     ECHELLES_GEO = type_geo,
                     NOM_TABLEAU = label_file,
                     ANNEE_GEOGRAPHIE = annee_geographie_DEP),
            by = "TABLEAU") %>%
  filter(!TABLEAU %in% 'Tous les tableaux') %>%
  select(TABLEAU, NOM_TABLEAU, VAR_COD = VAR_N, VAR_LIB = LIB, ANNEE_DONNEE, ECHELLES_GEO, ANNEE_GEOGRAPHIE)

# imports et aggrégation de tous les datasets du thème REC


argList <- list(liste_datasets$id_theme,
                liste_datasets$name_theme,
                liste_datasets$id_dataset, 
                liste_datasets$annee_recensement, 
                liste_datasets$annee_geographie_DEP)

data_SGF <- pmap_dfr(argList,
                    f_dl_import_md_dataset) %>%
  mutate(VAR_COD = as.numeric(str_replace_all(VAR_COD, "V","")))


# # Save an object to a file
# saveRDS(data_SGF, file = "./data/data_SGF.rds")
# # Restore the object
# data_SGF <- readRDS(file = "./data/data_SGF.rds")





################################
#### modifs pour intégration data arrondissements
################################

# modifs année géographie
indicateurs_SGF_NEW <- 
  indicateurs_SGF %>%
  mutate(ANNEE_GEOGRAPHIE = case_when(TABLEAU %in% 'MVTPOP_T91' ~ 1826, TRUE ~ ANNEE_GEOGRAPHIE))



# modifs année géographie
data_SGF_NEW <- data_SGF %>%
  mutate(NIVGEO = case_when(SRC_DATA %in% 'MVTPOP_T91' ~ "ARR", TRUE ~ as.character(NIVGEO))) %>%
  mutate(ANNEE_GEOGRAPHIE = case_when(SRC_DATA %in% 'MVTPOP_T91' ~ 1826, TRUE ~ ANNEE_GEOGRAPHIE))

# corrections arrondissements PARIS
data_SGF_NEW <- data_SGF_NEW %>%
  ungroup() %>%
  mutate(CODGEO_NEW = case_when(substr(CODGEO,1,3) %in% c('754','755','756') ~ "7501", TRUE ~ as.character(CODGEO))) %>%
  group_by(NIVGEO, CODGEO_NEW , SRC_DATA ,VAR_COD , VAR_LIB , ANNEE_DONNEE , ANNEE_GEOGRAPHIE) %>%
  summarise(LIBGEO = first(LIBGEO),
            VAL = sum(VAL, na.rm = T)) %>%
  select("NIVGEO", CODGEO = "CODGEO_NEW" , "LIBGEO" , "SRC_DATA" ,"VAR_COD" , "VAR_LIB" , "ANNEE_DONNEE" ,    "ANNEE_GEOGRAPHIE", "VAL") %>%
  mutate(LIBGEO = case_when(LIBGEO %in% 'LOUVRE' ~ "PARIS", TRUE ~ as.character(LIBGEO)))


data_SGF <- data_SGF_NEW
data_SGF <- data_SGF %>% ungroup() %>% as.data.frame()


## rectifs manuelles indicateurs_SGF
# suppr data/indicateurs_sgf.rda
indicateurs_SGF <- indicateurs_SGF %>%
  mutate(ECHELLES_GEO = case_when(TABLEAU %in% "MVTPOP_T76" ~ "CHL_VILLE",
                                  TRUE ~ as.character(ECHELLES_GEO)))
usethis::use_data( indicateurs_SGF, overwrite = T)


date_SGF <- data_SGF

data-raw/script_dev.R
library(usethis)
library(devtools)

#https://speakerdeck.com/colinfay/building-a-package-that-lasts-erum-2018-workshop
#https://www.hvitfeldt.me/blog/usethis-workflow-for-package-development/

unlink("DESCRIPTION")
my_desc <- desc::description$new("!new")
my_desc$set("Package","sgf")
my_desc$set("Title","sgf")
my_desc$set("Authors@R", "person('Observatoire des Territoires',email='observatoire@☺cget.gouv.fr', role=c('cre','aut'))")
#my_desc$set_authors()
my_desc$del("Maintainer")
my_desc$set_version("0.0.0.9000")
my_desc$set(Description = "Old French statistics")
my_desc$set("URL","https://github.com/observatoire-territoires/sgf")
my_desc$set("BugReports","https://github.com/observatoire-territoires/sgf/issues")

#use_gpl3_license()
use_mit_license(name = "Insee - SGF - ICPSR - Observatoire des territoires")

# save
my_desc$write(file = "DESCRIPTION")

# A FINIR
use_travis()

# ✔ Setting active project to 'C:/Users/mgarnier/Documents/diffusion_donnees/sgf'
# ✔ Writing '.travis.yml'
# ✔ Adding '^\\.travis\\.yml$' to '.Rbuildignore'
# ● Turn on travis for your repo at https://travis-ci.org/profile/observatoire-territoires
# ● Copy and paste the following lines into 'C:/Users/mgarnier/Documents/diffusion_donnees/sgf/README.Rmd':
#   <!-- badges: start -->
#   [![Travis build status](https://travis-ci.org/observatoire-territoires/sgf.svg?branch=master)](https://travis-ci.org/observatoire-territoires/sgf)
# <!-- badges: end -->
#   [Copied to clipboard]

# data-raw dir
devtools::use_data_raw()

# sauvegarde des dataframes dans dossier data
usethis::use_data(geo_DEP_SGF_histo, indicateurs_SGF, data_SGF, overwrite = T)
usethis::use_data( indicateurs_SGF, overwrite = T)
usethis::use_data( geo_ARR_SGF_histo, overwrite = T)

# rectifs des tables

usethis::use_data( indicateurs_SGF, data_SGF, overwrite = T)

usethis::use_data( data_SGF, overwrite = T)

usethis::use_data( geo_ARR_SGF_histo, overwrite = T)


use_news_md()

use_version()

# fonction
use_r("sgf_sfdf")

# documentation des datasets
use_r("data")

# dependances
use_package("dplyr") 
use_package("dplyr") 
use_package("tidyr") 
use_package("janitor") 
use_package("sf") 


# doc

library(pkgdown)
library(roxygen2)

#générer doc .Rd
roxygen2::roxygenise()

# vérification de l'intégrité du package
devtools::check()


data-raw/crea_carto.R
library(tidyverse)
library(sf)
library(rmapshaper)

# nettoyage puis création des mailles DEP
# par millésime

# 1801
geo_ARR_1801 <- st_read("./geo_source/arrondissements1801_S.shp") %>%
  st_transform(2154) %>%
  mutate(CODGEO = str_pad(IDEN, side = "left", width = 4, pad = "0")) %>%
  select(CODGEO)

geo_DEP_1801 <- geo_ARR_1801 %>% 
  mutate(CODGEO = substr(CODGEO,1,2)) %>%
  group_by(CODGEO) %>%
  summarise()

# 1826
geo_ARR_1826 <- st_read("./geo_source/arrondissements1826_S.shp") %>%
  st_transform(2154) %>%
  mutate(CODGEO = str_pad(IDEN, side = "left", width = 4, pad = "0")) %>%
  select(CODGEO)

geo_DEP_1826 <- geo_ARR_1826 %>% 
  mutate(CODGEO = substr(CODGEO,1,2)) %>%
  group_by(CODGEO) %>%
  summarise()

# 1866
geo_DEP_1866 <- geo_DEP_1876 %>% 
  filter(!CODGEO %in% c('90','57','54','88')) %>%
  rbind(geo_DEP_1826 %>% filter(CODGEO %in% c('57', '99','67','68','88'))) 


# 1876
geo_ARR_1876 <- st_read("./geo_source/arrondissements1876_S.shp") %>%
  st_transform(2154) %>%
  mutate(CODGEO = str_pad(IDEN, side = "left", width = 4, pad = "0")) %>%
  select(CODGEO)

geo_DEP_1876 <- geo_ARR_1876 %>% 
  mutate(CODGEO = substr(CODGEO,1,2)) %>%
  group_by(CODGEO) %>%
  summarise()


# 1918

geo_DEP_1918 <- st_read("https://raw.githubusercontent.com/gregoiredavid/france-geojson/master/departements.geojson") %>%
  rename(CODGEO = code, LIBGEO = nom) %>%
  st_transform(2154) %>%
  mutate(CODGEO = case_when(CODGEO %in% c('2A','2B') ~ '20',
                                #CODGEO %in% c('90') ~ '68',
                                CODGEO %in% c('75','92','93','94') ~ '75',
                                CODGEO %in% c('95','78','91') ~ '78',
                                TRUE ~ as.character(CODGEO))) %>%
  group_by(CODGEO) %>%
  summarise() %>%
  ms_simplify(keep = 0.01)



#################
## fichier unique de l'historique des geometries des départements

geo_DEP_SGF_histo <-
  geo_DEP_1801 %>% mutate(ANNEE_GEOGRAPHIE = 1801) %>%
  rbind.data.frame(geo_DEP_1826 %>% mutate(ANNEE_GEOGRAPHIE = 1826)) %>%
  rbind.data.frame(geo_DEP_1866 %>% mutate(ANNEE_GEOGRAPHIE = 1866)) %>%
  rbind.data.frame(geo_DEP_1876 %>% mutate(ANNEE_GEOGRAPHIE = 1876)) %>%
  rbind.data.frame(geo_DEP_1918 %>% mutate(ANNEE_GEOGRAPHIE = 1918))

# libellés départements

ref_lib_dep_1 <- data_SGF %>% filter(NIVGEO %in% 'DEP') %>% filter(SRC_DATA %in% 'REC_T01') %>% filter(VAR_COD %in% 10) %>%
  select(CODGEO, LIBGEO) %>%
  mutate(LIBGEO = case_when(CODGEO %in% '04' ~ "BASSES-ALPES",
                            CODGEO %in% '05' ~ "HAUTES-ALPES",
                            CODGEO %in% '31' ~ "HAUTE-GARONNE",
                            CODGEO %in% '43' ~ "HAUTE-LOIRE",
                            CODGEO %in% '52' ~ "HAUTE-MARNE",
                            CODGEO %in% '64' ~ "BASSES-PYRENEES",
                            CODGEO %in% '65' ~ "HAUTES-PYRENEES",
                            CODGEO %in% '67' ~ "BAS-RHIN",
                            CODGEO %in% '68' ~ "HAUT-RHIN",
                            CODGEO %in% '70' ~ "HAUTE-SAONE",
                            CODGEO %in% '74' ~ "HAUTE-SAVOIE",
                            CODGEO %in% '79' ~ "DEUX-SEVRES",
                            CODGEO %in% '87' ~ "HAUTE-VIENNE",
                            TRUE ~ as.character(LIBGEO)))


geo_DEP_SGF_histo <- 
  geo_DEP_SGF_histo %>%
  left_join(ref_lib_dep_1, by = "CODGEO") %>%
  mutate(LIBGEO = case_when(CODGEO %in% '54' ~ "MEURTHE-ET-MOSELLE",
                            CODGEO %in% '90' ~ "BELFORT",
                            TRUE ~ as.character(LIBGEO))) %>%
  select(CODGEO, LIBGEO, ANNEE_GEOGRAPHIE, geometry)


class(geo_DEP_SGF_histo) <- c("sf", "data.frame")

# save carto deps
#saveRDS(geo_DEP_SGF_histo, file = "./geo/geo_DEP_SGF_histo.rds")


##############################
#### fonds ARRONDISSEMENTS
#############################


geo_ARR_SGF_histo <-
  geo_ARR_1801 %>% mutate(ANNEE_GEOGRAPHIE = 1801) %>%
  rbind.data.frame(geo_ARR_1826 %>% mutate(ANNEE_GEOGRAPHIE = 1826)) %>%
  rbind.data.frame(geo_ARR_1876 %>% mutate(ANNEE_GEOGRAPHIE = 1876)) 


# correction des arrondissements avec iles

geo_ARR_SGF_histo <- geo_ARR_SGF_histo %>% group_by(CODGEO, LIBGEO, ANNEE_GEOGRAPHIE) %>% summarise()


# rectification du code et de l'arrondissement de Toul en 1876

geo_ARR_SGF_histo_2 <- geo_ARR_SGF_histo
class(geo_ARR_SGF_histo_2) <- c('sf','data.frame')

geo_ARR_SGF_histo_2 <-geo_ARR_SGF_histo_2  %>%
  ungroup() %>%
  mutate(CODGEO = case_when(CODGEO %in% '5404' & ANNEE_GEOGRAPHIE %in% 1876 ~ "5405",
         TRUE ~ as.character(CODGEO))) %>%
  mutate(LIBGEO = case_when(CODGEO %in% '5405' & ANNEE_GEOGRAPHIE == 1876 ~ "TOUL",
         TRUE ~ as.character(LIBGEO)))

# changement de format DEP sf
geo_DEP_SGF_histo_2 <- geo_DEP_SGF_histo
class(geo_DEP_SGF_histo_2) <- c('sf','data.frame')

# libellés départements

ref_lib_arr_1 <- data_SGF %>% 
  filter(NIVGEO %in% 'ARR') %>%
  # filter(SRC_DATA %in% 'REC_T01') %>%
  # filter(VAR_COD %in% 10) %>%
  distinct(CODGEO, LIBGEO) %>%
  filter(!LIBGEO %in% c('VILLEFR-DE-ROUERG','TARASCON',"PONT L'EVEQUE", "CHATILLON-SUR-SEI",
                        "NOGENT LE ROTROU","VILLEFRANC DE LAU","ISSODUN","SAVENAY","VILLENEUV-SUR-LOT",
                        "SCHELESTADT","GUEBWILLER","VILLE-FR-SUR-SAON", "VILLEFR-SUR-SAONE",
                        "SAINT-JEAN-DE-MAU","ST-JEAN-DE-MAUR","SAINT-JULLIEN","NEUFCHATEL-EN-BRA",
                        "LAROCHE-SUR-YON","LES SABLES-D'OLONNE","MOULINS-SUR-ALLIE") ) %>%
  mutate(LIBGEO = case_when(CODGEO %in% '6802' ~ 'BELFORT', TRUE ~ as.character(LIBGEO))) %>%
  filter(!substr(CODGEO,1,3) %in% c('754','755','756')) %>%
  distinct(CODGEO, LIBGEO)




geo_ARR_SGF_histo <- 
  geo_ARR_SGF_histo %>%
  left_join(ref_lib_arr_1, by = "CODGEO") %>%
  select(CODGEO, LIBGEO, ANNEE_GEOGRAPHIE, geometry)


class(geo_ARR_SGF_histo) <- c("sf", "data.frame")
